src/libpmem/meson.build

```diff
@@ -0,0 +1,33 @@
+sources = files(
+    'libpmem.c',
+    '../libpmem2/memops_generic.c',
+    'pmem.c',
+    'pmem_posix.c',
+)
+
+libpmem = library(
+    'pmem',
+    sources,
+    c_args: [
+        '-Wl,--version-script=@0@'.format(meson.current_source_dir() / 'libpmem.link.in')
+    ],
+    dependencies: [
+        threads_dep,
+        libndctl_dep,
+        libdaxctl_dep,
+        pmemcommon_dep,
+        pmemcore_dep,
+        pmem2arch_dep,
+    ],
+    include_directories: root_include,
+    soversion: '1',
+    version: '1.0.0',
```

Comment:
If soversion is unspecified but version is provided, it defaults to the first component of version. So you can probably just skip explicitly listing the soversion. :)